src/scheduler.rs

```diff
@@ -173,17 +183,22 @@ mod tests {
 
     use crate::config_waiter::BootResult;
     use crate::manager::build_progress_key;
+    use crate::types::shared::RegionScheduleMode;
     use crate::{
         config_store::ConfigStore,
         types::{
             check_config::CheckConfig,
             result::{CheckResult, CheckStatus},
         },
     };
+    static TEST_MUTEX: Mutex<()> = Mutex::new(());
 
     #[traced_test]
     #[tokio::test(start_paused = true)]
     async fn test_scheduler() {
+        // We don't want these tests to run at the same time, since they access the external redis.
+        let _guard = TEST_MUTEX.lock().unwrap();
```

Comment:
I wonder if there's actually a better way to do this. Let me post in the rust channel